### PR TITLE
refactor: move nuxt plugin from composables to theme

### PR DESCRIPTION
**Pull request by ghost**
- [x] Use nuxt-universal cookie instead of nuxt/cookie.js 
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/modules/magento/defaultConfig.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const defaultConfig : Record<string, any> = {
```
Comment (sethidden) [inline packages/theme/modules/magento/defaultConfig.ts:1]: I've rewritten the .js files to .ts

File: packages/theme/modules/magento/plugin.ts [hunk 2]
```diff
@@ -0,0 +1,62 @@
+import { integrationPlugin } from '~/helpers/integrationPlugin';
+import { mapConfigToSetupObject } from '~/modules/magento/helpers';
+import { defaultConfig } from '~/modules/magento/defaultConfig';
+
+const moduleOptions = JSON.parse('<%= JSON.stringify(options) %>');
+
+// TODO should be moved to THEME and expose consistent cookie management API
```
Comment (sethidden) [inline packages/theme/modules/magento/plugin.ts:7]: @Frodigo @bartoszherba please let me know what further needs to be done in regards to to the `expose consistent cookie management API` part. Seems out of scope for this task but you decide during the review

**Review by RetroModernDev — CHANGES_REQUESTED** [review]
Looks good, added one comment about cookie consistency.

File: packages/theme/modules/magento/plugin.ts [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+export default integrationPlugin((plugin) => {
+  const cartCookieName : string = moduleOptions.cookies?.cartCookieName || defaultConfig.cookies.cartCookieName;
+  const customerCookieName : string = moduleOptions.cookies?.customerCookieName || defaultConfig.cookies.customerCookieName;
+  const storeCookieName : string = moduleOptions.cookies?.storeCookieName || defaultConfig.cookies.storeCookieName;
+  const currencyCookieName : string = moduleOptions.cookies?.currencyCookieName || defaultConfig.cookies.currencyCookieName;
+  const localeCookieName : string = moduleOptions.cookies?.localeCookieName || defaultConfig.cookies.localeCookieName;
+  const countryCookieName : string = moduleOptions.cookies?.countryCookieName || defaultConfig.cookies.countryCookieName;
+
+  const { $cookies } = plugin.app;
```
[the remark this answers is not in the document]
Comment (sethidden) [reply packages/theme/modules/magento/plugin.ts:16]: Ok, I'll change the $cookies references to `context.$magento.config.state.setStore(StoreConfigGetters.getCode(store));` etc.

[the remark this answers is not in the document]
Comment (sethidden) [reply packages/theme/modules/magento/plugin.ts:16]: On second thought - as discussed on Slack - I'll do it in a separate task/PR.

[the remark this answers is not in the document]
Comment (sethidden) [reply packages/theme/modules/magento/plugin.ts:16]: https://vsf.atlassian.net/browse/M2-331

File: packages/theme/modules/magento/plugin.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  const getCountry = () => $cookies.get(countryCookieName);
+  const setCountry = (country: string) => (!country ? $cookies.remove(countryCookieName) : $cookies.set(countryCookieName, country));
+
+  const settings = mapConfigToSetupObject({
+    moduleOptions,
+    app: plugin.app,
+    additionalProperties: {
+      state: {
+        // Cart
```
Comment (bartoszherba) [inline packages/theme/modules/magento/plugin.ts:41]:
```suggestion
```

File: packages/theme/modules/magento/plugin.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        getCartId,
+        setCartId,
+        // Customer
```
Comment (bartoszherba) [inline packages/theme/modules/magento/plugin.ts:44]:
```suggestion
```

File: packages/theme/modules/magento/plugin.ts [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        getCustomerToken,
+        setCustomerToken,
+        // Store
```
Comment (bartoszherba) [inline packages/theme/modules/magento/plugin.ts:47]:
```suggestion
```

**Review by bartoszherba — APPROVED** [review]

**Review by RetroModernDev — APPROVED** [review]

**Pull request merged**